ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -314,14 +292,66 @@ func convertToContainerDef(inputCfg *adapter.ContainerConfig, ecsContainerDef *C
 		outputContDef.LinuxParameters.SetTmpfs(inputCfg.Tmpfs)
 	}
 
-	// Set fields from ecs-params file
+	// initialize container resources from inputCfg
+	cpu := inputCfg.CPU
+	mem := inputCfg.Memory
+	memRes := inputCfg.MemoryReservation
+
+	// Set essential & resource fields from ecs-params file, if present
 	if ecsContainerDef != nil {
 		outputContDef.Essential = aws.Bool(ecsContainerDef.Essential)
+
+		cpu = getResourceValue(inputCfg.Name, cpu, ecsContainerDef.Cpu)
+
+		ecsMemInMB := adapter.ConvertToMemoryInMB(int64(ecsContainerDef.Memory))
+		mem = getResourceValue(inputCfg.Name, mem, ecsMemInMB)
+
+		ecsMemResInMB := adapter.ConvertToMemoryInMB(int64(ecsContainerDef.MemoryReservation))
+		memRes = getResourceValue(inputCfg.Name, memRes, ecsMemResInMB)
+	}
+
+	// One or both of memory and memoryReservation is required to register a task definition with ECS
+	// If neither is provided by 1) compose file or 2) ecs-params, set default
+	// NOTE: Docker does not set a memory limit for containers
+	if mem == 0 && memRes == 0 {
+		mem = defaultMemLimit
+	}
+
+	if mem != 0 && memRes != 0 && mem < memRes {
+		return nil, errors.New("mem_limit must be greater than mem_reservation")
+	}
+
+	outputContDef.SetCpu(cpu)
+	if mem != 0 {
+		outputContDef.SetMemory(mem)
+	}
+	if memRes != 0 {
+		outputContDef.SetMemoryReservation(memRes)
 	}
 
 	return outputContDef, nil
 }
 
+func getResourceValue(serviceName string, inputVal, ecsVal int64) int64 {
+	if inputVal > 0 && ecsVal > 0 {
+		showResourceOverrideMsg(serviceName, inputVal, ecsVal)
+		return ecsVal
+	}
+	if ecsVal == 0 {
+		return inputVal
+	}
+	return ecsVal
+}
```

Comment:
This is great! Though I _think_ this can be even further simplified to:
```
func getResourceValue(serviceName string, inputVal, ecsVal int64) int64 {
	if ecsVal == 0 {
		return inputVal   // inputVal can be 0 since we check for this later
	}
	if inputVal > 0 {  // we know ecsVal must be greater than 0 at this point so no need to include in condition
		showResourceOverrideMsg(serviceName, inputVal, ecsVal)
	}
	return ecsVal
}
```